WISE-Community/WISE-Client
# feat(Branching): Create edit branch view

## Changes
- Created edit branch view
- Created AbstractBranchAuthoringComponent that EditBranchComponent and AddBranchComponent both extend

Note: the edit branch submit button does not actually change the project yet

## Test
- Add Branch
  - Make sure the add branch view still works
- Edit Branch
  - In the project view, click the branch icon on a step to open the edit branch view
  - Make sure the edit branch view populates the branch parameters
  - Make sure the "Submit" button alerts the new branch parameters
  - Make sure the "Remove Branch" button actually removes the branch structure and changes the project
